config/config.go

```diff
@@ -0,0 +1,84 @@
+package config
+
+import (
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+)
+
+// RawConfig - used to unmarshall the raw config file
+type RawConfig struct {
+	S3 rawResourceType `yaml:"s3"`
+}
+
+type rawResourceType struct {
+	IncludeRule rawFilterRule `yaml:"include"`
+	ExcludeRule rawFilterRule `yaml:"exclude"`
+}
+
+type rawFilterRule struct {
+	NamesRE []string `yaml:"names_regex"`
+}
+
+// Config - the config object we pass around
+// that is a parsed version of RawConfig
+type Config struct {
+	S3 ResourceType
+}
+
+// ResourceType - the include and exclude
+// rules for a resource type
+type ResourceType struct {
+	IncludeRule FilterRule
+	ExcludeRule FilterRule
+}
+
+// FilterRule - contains regular expressions or plain text patterns
+// used to match against a resource type's properties
+type FilterRule struct {
+	NamesRE []*regexp.Regexp
+}
+
+// GetConfig - unmarshall the raw config file
+// and parse it into a config object.
+func GetConfig(filePath string) (*Config, error) {
+	var configObj Config
+
+	absolutePath, err := filepath.Abs(filePath)
+	if err != nil {
+		return &Config{}, err
```

Comment:
NIT: We typically do `return nil, err` in these cases.

Reply:
- [ ] TODO